elytron/WFLY-19314-oidc-logout-support.adoc

```diff
@@ -0,0 +1,191 @@
+as specified in the https://openid.net/specs/openid-connect-core-1_0.html[OpenID Connect Core 1.0 spec].
+
+There are additional optional OIDC specifications that may be used in addition to the Core spec to
+provide support for logging out:
+
+* https://openid.net/specs/openid-connect-rpinitiated-1_0.html[RP-Initiated Logout] is an optional
+OIDC specification that defines how a relying party (RP) can request that an OpenID provider logs
+out the end-user.
+* https://openid.net/specs/openid-connect-frontchannel-1_0.html[Front-Channel Logout] is an optional
+OIDC specification that defines a front-channel logout mechanism that does not use an OP iframe on RP
+pages.
+* https://openid.net/specs/openid-connect-core-1_0.html#OpenID.BackChannel[Back-channel Logout] is an
+optional OIDC specification that defines a logout mechanism that uses direct back-channel communication
+between the OP and RPs being logged out.
+
+This RFE is to add support for these three types of logout mechanisms when securing apps deployed to
+WildFly with OIDC.
+
+=== User Stories
+* A gaming portal uses wildfly to provide customers access to select games, a gaming email
+    and chat app.  When the portal detects a user's session has expired for non-activity
+    the user is logged out of all applications opened during the session,
+    (OIDC backchannel-logout).
+
+* A gaming portal uses wildfly to provide customers access to select games, email,
+    and chat.  When the user logs out of the portal all currently open applications
+    are also logged out, (OIDC frontchannel-logout).
+
+== Issue Metadata
+
+=== Related Issues
+* https://issues.redhat.com/browse/ELY-2534[OIDC logout code implementation in wildfly-elytron]
+* https://issues.redhat.com/browse/WFLY-19959[Increment (preview) version of wildfly-elytron-oidc-client.xsd]
+* https://issues.redhat.com/browse/WFLY-19960[Add OIDC logout configuration elements]
+* https://issues.redhat.com/browse/WFLY-19314[testsuite tests for logout and documentation in docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc]
+*TODO wildfly-core jira for wildfly-elytron version update
+
+=== Affected Projects or Components
+
+* WildFly Elytron
+* WildFly Core
+* WildFly
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+* It will be possible to log out of an application secured with OIDC using RP-Initiated Logout, Front-Channel Logout,
+and Back-Channel Logout.
+* The new configuration options described below will be added for both the `elytron-oidc-client` subsystem configuration
+and deployment configuration.
+* This feature will be available when starting WildFly with the Preview Stability Level.
```

Comment:
Will the feature be enabled by default? If yes, will it be possible to turn it off?

Reply:
Description added